src/masonite/drivers/session/RedisDriver.py

```diff
@@ -0,0 +1,147 @@
+import json
+from typing import TYPE_CHECKING
+
+from ..BaseDriver import BaseDriver
+from ...validation import MessageBag
+
+if TYPE_CHECKING:
+    from ...foundation import Application
+
+
+class RedisDriver(BaseDriver):
+    """Session driver used to store data in Redis."""
+
+    def __init__(self, application: "Application"):
+        super().__init__(application)
+        self.connection = None
+        self.driver_config = {"decode_responses": True}
+
+    def get_connection(self):
+        try:
+            import redis
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Could not find the 'redis' library. Run 'pip install redis' to fix this."
+            )
+
+        if not self.connection:
+            self.connection = redis.StrictRedis(
+                **self.driver_config,
+            )
+
+        return self.connection
+
+    def start(self) -> dict:
+        data = {}
+        flashed = {}
+
+        session_data = {}
+        cursor = '0'
+        session_prefix = self.get_session_namespace() + '*'
+        while cursor != 0:
+            cursor, keys = self.get_connection().scan(cursor=cursor, match=session_prefix, count=100000)
+            if keys:
+                values = self.get_connection().mget(*keys)
+                # values = map(int, values)
+                session_data.update(dict(zip(keys, values)))
+
+        data_prefix = self.get_data_prefix()
+        flash_prefix = self.get_flash_prefix()
+        for key, value in session_data.items():
+            if key.startswith(data_prefix):
+                data.update({key.replace(data_prefix, ""): value})
+            elif key.startswith(flash_prefix):
+                flashed.update({key.replace(flash_prefix, ""): value})
+
+        return {"data": data, "flashed": flashed}
+
+    def save(
+        self, added=None, deleted=None, flashed=None, deleted_flashed=None
+    ) -> None:
+        if added is None:
+            added = {}
+        if deleted is None:
+            deleted = []
+        if flashed is None:
+            flashed = {}
+        if deleted_flashed is None:
+            deleted_flashed = []
+
+        for key, value in flashed.items():
+            if isinstance(value, (MessageBag)):
+                value = value.json()
+            self.set(f"{self.get_flash_prefix()}{key}", value)
+
+        for key, value in added.items():
+            self.set(f"{self.get_data_prefix()}{key}", value)
+
+        for key in deleted:
+            self.delete(f"{self.get_data_prefix()}{key}")
+
+        for key in deleted_flashed:
+            self.delete(f"{self.get_flash_prefix()}{key}")
+
+    def get_session_namespace(self, session_id=None):
+        if not session_id:
+            request = self.application.make("request")
+            session_id = request.cookie("SESSID")
+        ns = self.options.get("namespace", "")
```

Comment:
ns variable should be called `namespace`. Remember code is read more than its written. No need for shorthanded variable names

Reply:
Fixed